File: src/etch/bindings/java/msg/test/TestField.java
/* $Id$
 *
 * Created by sccomer on Feb 4, 2007.
 *
 * Copyright (c) 2007 Cisco Systems, Inc. All rights reserved.
 */

package etch.bindings.java.msg.test;

import static org.junit.Assert.assertEquals;

import org.junit.Test;

import etch.bindings.java.msg.Field;
import etch.bindings.java.msg.IdName;

public class TestField
{
	@Test
	public void fieldIntegerString()
	{
		testMf( 1, "one" );
		testMf( 2, "two" );
		testMf( 3, "three" );
	}

	@Test
	public void fieldString()
	{
		testMf( "one" );
		testMf( "two" );
		testMf( "three" );
	}

	private void testMf( int id, String name )
	{
		Field mf = new Field( id, name );
		assertEquals( id, mf.getId() );
		assertEquals( name, mf.getName() );
	}

	private void testMf( String name )
	{
		Field mf = new Field( name );
		assertEquals( IdName.hash( name ), mf.getId() );
		assertEquals( name, mf.getName() );
	}
}

File: src/etch/bindings/csharp/msg/test/TestField.cs
///$Id$
///
///Created by Champakesan, Badri Narayanan on Jun 8, 2007.
///
///Copyright (c) 2007 Cisco Systems, Inc. All rights reserved.


using System;
using NUnit.Framework;

namespace Etch.Msg.Test
{

    [TestFixture]
    public class TestField
    {
        [TestFixtureSetUp]
        public void First()
        {
            Console.WriteLine();
            Console.Write( "TestField" );
        }

        [Test]
        public void FieldIntegerString()
        {
            TestMf(1, "one");
            TestMf(2, "two");
            TestMf(3, "three");
        }

        [Test]
        public void FieldString()
        {
            TestMf("one");
            TestMf("two");
            TestMf("three");
        }

        private void TestMf(int id, string name)
        {
            Field mf = new Field(id, name);
            Assert.AreEqual(id, mf.Id);
            Assert.AreEqual(name, mf.Name);
        }

        private void TestMf(string name)
        {
            Field mf = new Field(name);
            Assert.AreEqual(IdName.Hash(name), mf.Id);
            Assert.AreEqual(name, mf.Name);
        }
    }
}